### PR TITLE
fix(fm): altitude formatting

### LINKED ISSUE #8174: ND: LEMD TLD7A constraints showing as flight level and not altitude
### Aircraft Version

Development

### Build info

```json
{
    "built": "2023-08-15T09:43:50+00:00",
    "ref": "refs/heads/master",
    "sha": "a285b2aa4fc9e88472ade82e3c44c2310c66faf0",
    "actor": "tracernz",
    "event_name": "manual",
    "pretty_release_name": "master:a285b2aa",
    "version": "v0.11.0-dev.a285b2a"
}
```

### Describe the bug

TLD7A altitude constraints appear as flight levels despite being lower than the transition level and published as altitudes on the charts.
CSTR1
CSTR2
CSTR3

### Expected behavior

Constraints to appear as an altitude instead of a level when such criteria is applicable.

### Steps to reproduce

1) Load LEMD TLD7A to 18R arrival and observe the altitude constraints 

### References (optional)

_No response_

### Additional info (optional)

Using navigraph navdata for msfs.

### Discord Username (optional)

bruno_pt99

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #8174

## Summary of Changes

Corrects altitude/flight level formatting with respect to transition altitude and transition level. For constraints we explicitly know their type so we can pick the correct one, but for other altitudes on the MCDU F-PLN page we need to try to guess because VNAV does not provide us this information.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Test some procedures with the transition altitude (PERF TO) and transition level (PERF APPR, labelled as TRANS ALT) set to different altitudes. Make sure the trans alt only affects climb altitudes and the transition level only affects cruise and descent altitudes. See the fixed issue for more context if needed.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js [hunk 1]
```diff
@@ -198,6 +219,16 @@ class CDUFlightPlanPage {
                 }
             }
 
+            const constraintType = wp ? CDUVerticalRevisionPage.constraintType(mcdu, wp) : WaypointConstraintType.Unknown;
+            if (constraintType === WaypointConstraintType.CLB) {
+                useTransitionAltitude = true;
+            } else if (constraintType === WaypointConstraintType.DES) {
+                useTransitionAltitude = false;
+            } else if (tocIndex >= 0) {
+                // FIXME Guess because VNAV doesn't tell us whether altitudes are climb or not \o/
```
Comment (BlueberryKing) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:228]: soon™

**Review by BlueberryKing — APPROVED** [review]
LGTM

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js (hunk 1 above)
Comment (tracernz) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_FlightPlanPage.js:228]: There's a reason I pinged you for review. 😁 I'd appreciate if you can follow this one up when you have a chance some time 👍 .

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing: #8186 
Tier of Testing : 1
Date : 02/09/2023

Testing Process:
- Fly LGSK (transition 5000) -> LGAV (transition 10000)
- Observe transition altitude changing to FL depending on state 
- [PASS] For CLB 
- [PASS] For APPR

Testing Results:
Passed

**Pull request merged**